app/models/service_retire_task.rb

```diff
@@ -26,40 +26,40 @@ def after_request_task_create
     parent_svc = Service.find_by(:id => options[:src_ids])
     _log.info("- creating service tasks for service <#{self.class.name}:#{id}>")
 
-    create_retire_subtasks(parent_svc)
+    create_retire_subtasks(parent_svc, self)
   end
 
-  def create_retire_subtasks(parent_service)
-    parent_service.direct_service_children.each { |child| create_retire_subtasks(child) }
+  def create_retire_subtasks(parent_service, parent_task)
     parent_service.service_resources.collect do |svc_rsc|
       next unless svc_rsc.resource.try(:retireable?)
       # TODO: the next line deals with the filtering for provisioning
       # (https://github.com/ManageIQ/manageiq/blob/3921e87915b5a69937b9d4a70bb24ab71b97c165/app/models/service_template/filter.rb#L5)
       # which should be extended to retirement as part of later work
       # svc_rsc.resource_type != "ServiceTemplate" || self.class.include_service_template?(self, srr.id, parent_service)
-      nh = attributes.except("id", "created_on", "updated_on", "type", "state", "status", "message")
+      nh = attributes.except("id", "created_on", "updated_on", "type", "state", "status", "message", *self.class.virtual_attribute_names)
```

Comment:
Just an alternative to @gmcculloug 's suggestion:

```ruby
nh = attributes.slice(self.class.column_names)
               .except!("id", "created_on", "updated_on", "type", "state", "status", "message")
```

Just makes it so that the list you are starting with is always what is available as columns from the database, and then it excludes columns from there, in the rare case that someone has fetch this current instance of this object was fetched using a custom `SELECT` statement that included some other derived columns.

... Not that I would know anyone who would do that, for say, performance reasons or anything...

( ͡° ͜ʖ ͡°)

Note:  I just used `.except!` here to not dup the hash a second time, but that is probably peanuts in the grand scheme of things.